router-tests/compression_test.go

```diff
@@ -0,0 +1,112 @@
+package integration_test
+
+import (
+	"compress/flate"
+	"compress/gzip"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	br "github.com/andybalholm/brotli"
+	"github.com/go-chi/chi/v5"
+	"github.com/go-chi/chi/v5/middleware"
+)
+
+func UnitTestCompression(t *testing.T) {
+	httpRouter := chi.NewRouter()
+	var customCompressibleContentTypes = []string{
```

Comment:
Unit tests should be in the router package. I recommend to create a compression package where you can define the code and variables.